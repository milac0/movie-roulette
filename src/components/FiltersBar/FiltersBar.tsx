import React, { useContext } from "react";
import { DataContext } from "../../context/DataContext";
import Filters from "./../../enums/Filters";
import styles from "./filtersBar.scss";

interface Props {}

const SearchBar: React.FC<Props> = () => {
  const { filterBy, setFilterBy, setQuery } = useContext(DataContext);

  const handleClick = (e: React.MouseEvent<HTMLButtonElement>) => {
    setQuery("");
    setFilterBy(e.currentTarget.value as Filters);
  };

  return (
    <div className={styles.container}>
      {Object.keys(Filters).map((key: keyof typeof Filters, i) => (
        <button
          key={i}
          onClick={handleClick}
          value={Filters[key]}
          className={filterBy === Filters[key] ? styles.activeFilter : null}
        >
          {key}
        </button>
      ))}
    </div>
  );
};

export default SearchBar;
