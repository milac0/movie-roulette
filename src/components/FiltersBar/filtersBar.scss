@import "../../theme/variables";

.container {
  background: $backgroundGrey;
  display: flex;
  justify-content: center;
  padding: 1em;

  button {
    @extend %sharedBtn;
  }

  .activeFilter {
    color: #fff;
    outline: none;
  }
}
