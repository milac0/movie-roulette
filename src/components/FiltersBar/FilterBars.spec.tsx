import React from 'react';
import renderer from 'react-test-renderer';

import FiltersBar from './FiltersBar';

describe('FiltersBar renders correctly', () => {
    test('snapshot renders', () => {
        const tree = renderer.create(<FiltersBar />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

