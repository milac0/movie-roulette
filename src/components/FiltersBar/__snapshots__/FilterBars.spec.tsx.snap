// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersBar renders correctly snapshot renders 1`] = `
<div
  className="container"
>
  <button
    className="activeFilter"
    onClick={[Function]}
    value="popular"
  >
    Popular
  </button>
  <button
    className={null}
    onClick={[Function]}
    value="now_playing"
  >
    Now Playing
  </button>
  <button
    className={null}
    onClick={[Function]}
    value="top_rated"
  >
    Top Rated
  </button>
  <button
    className={null}
    onClick={[Function]}
    value="upcoming"
  >
    Upcoming
  </button>
</div>
`;
